### PR TITLE
fix(telemetry): don't send header names on Studio if send_headers is None

### LINKED ISSUE #2403: When `send_headers` is set to None for usage reporting, do not send header names
For now we're sending header names with empty values if  `send_headers` is set to none. It's a mistake we should follow what [the gw is doing](https://www.apollographql.com/docs/apollo-server/api/plugin/usage-reporting#sendheaders) and don't send anything

### PR DESCRIPTION
**Pull request by bnjjj**
- Fix #2403

Before this PR is `send_headers`  was set to `none` we sent header names with empty header values. Now we don't send anything

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]. It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]. Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]. Tick whichever testing boxes are applicable. If you are adding Manual Tests:
    - please document the manual testing (extensively) in the Exceptions.
    - please raise a separate issue to automate the test and label it (or ask for it to be labeled) as `manual test`

### CODE REVIEW

File: NEXT_CHANGELOG.md [hunk 1]
```diff
@@ -115,6 +115,13 @@ By [@Geal](https://github.com/geal) in https://github.com/apollographql/router/p
 
 ## 🐛 Fixes
 
+### Don't send header names on Studio if `send_headers` is `none` ([Issue #2403](https://github.com/apollographql/router/issues/2403))
```
Comment (BrynCooke) [inline NEXT_CHANGELOG.md:118]:
```suggestion
### Don't send header names to Studio if `send_headers` is `none` ([Issue #2403](https://github.com/apollographql/router/issues/2403))
```

File: NEXT_CHANGELOG.md [hunk 2]
```diff
@@ -115,6 +115,13 @@ By [@Geal](https://github.com/geal) in https://github.com/apollographql/router/p
...
+
+Before when `send_headers` was set to `none` we sent header names with empty header values. Now we don't send anything to Studio.
+
```
Comment (BrynCooke) [inline NEXT_CHANGELOG.md:121]: Suggest adding minimal example yaml config.

**Review by BrynCooke — APPROVED** [review]
Approved with a couple of suggstions.

File: NEXT_CHANGELOG.md (hunk 1 above)
Comment (bnjjj) [reply NEXT_CHANGELOG.md]: Fixed in https://github.com/apollographql/router/pull/2425/commits/bb88f4b716998620d75109ff75731e25b9859709

File: NEXT_CHANGELOG.md (hunk 2 above)
Comment (bnjjj) [reply NEXT_CHANGELOG.md:127]: Fixed in https://github.com/apollographql/router/pull/2425/commits/bb88f4b716998620d75109ff75731e25b9859709

**Pull request merged**